ground/src/main/java/com/google/android/ground/ui/map/gms/GoogleMapsFragment.kt

```diff
@@ -209,15 +211,16 @@ class GoogleMapsFragment : SupportMapFragment(), MapFragment {
     }
   }
 
-  private fun onMarkerClick(marker: Marker): Boolean =
-    if (getMap().uiSettings.isZoomGesturesEnabled) {
-      markers[marker]?.let { markerClicks.onNext(it) }
+  private fun onClusterItemClick(item: LocationOfInterestClusterItem): Boolean {
```

Comment:
Seems this is replacing `onMarkerClick, but we'll still need a way to handle these.

Reply:
This is handling marker clicks; the cluster manager becomes responsible for managing clicks on map markers. It's a bit confusing from the terminology, but this is equivalent to asking it to handle a single click on an individual marker ("LOI Item"). I can refactor the name back to "on marker click" if that's clearer.

Reply:
Ah ok, then we can leave it as-is for now. Maybe please just add a Javadoc explaining that this method will handle both normal ping and cluster marker clicks.